### PR TITLE
Feature: Funfacts

### PR DESCRIPTION
**Pull request by j10a1n15**
Required:
- https://github.com/PartlySaneStudios/partly-sane-skies-public-data/pull/93

### CODE REVIEW

**Review by Su386yt — CHANGES_REQUESTED** [review]
Missing documentation

File: src/backend/FunFactData.ts [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+// See LICENSE for copyright and license notices.
+//
+
+import { getData } from '../utils/SystemUtils';
+import { prisma } from './backend';
+import { CronJob } from 'cron';
+
+const REPO = "partly-sane-skies-public-data";
+const OWNER = "j10a1n15";
+const FILE_PATH = "data/constants/fun_facts.json";
```
[the remark this answers is not in the document]
Comment (j10a1n15) [reply src/backend/FunFactData.ts:12]: https://github.com/PartlySaneStudios/partly-sane-skies-public-data/pull/91

File: src/backend/FunFactData.ts [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+// Written by J10a1n15.
+// See LICENSE for copyright and license notices.
+//
+
+import { getData } from '../utils/SystemUtils';
+import { prisma } from './backend';
+import { CronJob } from 'cron';
+
+const REPO = "partly-sane-skies-public-data";
+const OWNER = "j10a1n15";
```
[the remark this answers is not in the document]
Comment (j10a1n15) [reply src/backend/FunFactData.ts:11]: i dont think I wanna switch away from pss repo anyway lol

File: src/backend/FunFactData.ts [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+            console.log(`Daily fun fact updated for ${new Date().toLocaleDateString()}`);
+        }
+    } catch (error) {
+        console.error("Error updating daily fun fact:", error);
+    }
+}
+
+export async function loadFunFactData(override: boolean = false): Promise<void> {
+    const funFacts = await prisma.funFact.findMany();
+    if (funFacts.length == 0 || override) {
```
Comment (j10a1n15) [inline src/backend/FunFactData.ts:83]: Clear the old funfacts of you want to override

**Pull request merged**